apps/sparo-lib/src/services/GitService.ts

```diff
@@ -445,6 +446,59 @@ Please specify a directory on the command line
     return currentBranch;
   }
 
+  /**
+   * Retrieves the previous branch name using `@{-n}` syntax
+   *
+   * Assume:
+   * git checkout feature
+   * git checkout main
+   * ---
+   * `git checkout @{-1}` equals to run `git checkout feature`.
+   * Running `getPreviousBranch(1)` works in the similar way and returns "feature" in this case.
+   */
+  public getPreviousBranch(n: number): string | undefined {
```

Comment:
instead of mannualy parse git logs, why not considering of use `git rev-parse --symbolic-full-name --abbrev-ref=loose @{-n}`

Reply:
That's neat. Let me refactor this function with this suggested git command.